### PR TITLE
MM-975 Added timestamps to pictures to stop caching

### PR DESCRIPTION
**Pull request by nickago**

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ type PostStore interface {
 type UserStore interface {
 	Save(user *model.User) StoreChannel
 	Update(user *model.User, allowRoleUpdate bool) StoreChannel
+	UpdateUpdateAt(userId string) StoreChannel
```
Comment (crspeller) [inline store/store.go]: Why have you added this if you do not call it anywhere?

Comment (crspeller) [inline store/store.go]: @nickago I have not merged this because I am waiting for a response.

Comment (nickago) [inline store/store.go]: Yah, I'm working on refactoring due to a bug that came up I originally didn't see. Should be ready soon.

**Comment by nickago** [comment]
Alright, ready for re-review

**Comment by ralder** [comment]
sorry, but why not using etag header for cache strategy in `api/user.go` `func getProfileImage`

**Comment by crspeller** [comment]
Did etags not work for this? 
Also if I update my profile (but not my picture) the profile picure's time stamp will not but updated but the users will. Will that cause the display of the profile picture to break?

**Comment by nickago** [comment]
Etags was my original strategy, but it failed to update the way I wanted it to. I spoke with @coreyhulen about this, and we determined some combination of the browsers internal caching system and the bare url was causing them not to update.
As for the time stamp, no, updating the user profile will not invalidate the picture. The time stamp is there to invalidate the browser's link cache and force it to go to the server to get the new one, not to be checked against the picture.

**Comment by crspeller** [comment]
Ok looks good then.

**Pull request merged**